packages/blazor-workspace/Tests/BlazorWorkspace.Testing.Acceptance/AcceptanceTestsBase.cs

```diff
@@ -33,6 +33,9 @@ private async Task NavigateToPageAsync(IPage page, string route)
     private async Task WaitForComponentsInitializationAsync(IPage page)
     {
         await page.WaitForFunctionAsync(ComponentLibraryInitializationTestJavaScript);
+        // Have encountered some test intermittency when interacting with the components immediately
+        // after the JS scripts report ready. Added an extra wait here to mitigate.
+        await Task.Delay(150);
```

Comment:
This delay is unfortunate. Do we have any path out of this? Could we file an issue to Blazor / Playwright about it? What else do we know about the lifecycle? Is something specific not available yet so only certain tests might need this?

Reply:
I'll do some further investigation.

Reply:
The following comment may help avoid the wait (and inline js): https://github.com/ni/nimble/pull/2454#discussion_r1821424713

Reply:
At least locally, I've only seen wafer map test failures after removing the Task.Delay (with my other most recent changes).

It doesn't look like those tests were waiting for the wafer map to render. (Some tests had an unused 'render text field' that was probably the start of code intended for that, but unfinished.) Updated the tests to wait for the wafer map render (setDataAsync) to be done. Pushed those changes so I can get some runs on the CI to see if any other failures pop up.

Reply:
It looks like the tests are passing now without the delay (I'll keep monitoring subsequent builds though).

I think the other piece of the puzzle was [prerendering](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#prerendering) - on by default for interactive render modes, components start out rendered by the server, without event handlers hooked up. But from the Playwright/browser perspective, they'd look ready for interactions.

Updated the acceptance test pages to opt out of prerendering. Will resolve this thread once I update the docs too.

Reply:
Updated docs